### PR TITLE
Move restmapper from api/meta to api/rest/restmapper

### PR DESCRIPTION
**Pull request by zhengguoyong**
This change is [Reviewable](http://reviewable.k8s.io/reviews/kubernetes/kubernetes/17824)

### CODE REVIEW

File: pkg/api/meta/meta.go [hunk 1]
```diff
@@ -104,13 +104,13 @@ func TypeAccessor(obj interface{}) (TypeInterface, error) {
 // or manipulate resource version on objects derived from core API
 // metadata concepts.
 func NewAccessor() MetadataAccessor {
-	return resourceAccessor{}
+	return ResourceAccessor{}
```
Comment (smarterclayton) [inline pkg/api/meta/meta.go]: Why does this need to be public?

Comment (zhengguoyong) [inline pkg/api/meta/meta.go]: Because package restmapper(pkg/api/rest/restmapper/restmapper_test.go) will use it. :)

**Comment by deads2k** [comment]
I don't have any real objection to moving the package (`meta` was a bad fit anyway), but is this part of some larger plan or issue that you could link to?

Also, the new package `pkg/api/rest/restmapper` isn't an obvious choice to me.  Has there been discussion about where we want the package to live?  It seems very "client-y" to me.  

/cc @caesarxuchao @krousey

**Comment by caesarxuchao** [comment]
> It seems very "client-y" to me.

Currently pkg/apiserver also imports the RESTMapper, though perhaps it shouldn't.

> is this part of some larger plan or issue that you could link to?

I don't know any plan on where to move the RESTMapper. I know @krousey is refactoring the client-side RESTMapper mechanism, so maybe he'll chime in.

**Comment by lavalamp** [comment]
@zhengguoyong Sorry this aged for so long. I'm going to assign to @krousey since he's potentially making changes to this in the nearish future.

**Pull request closed without merging**